### DISCUSSION TITLE
How to add authorization documentation to printed schema

### DISCUSSION THREAD

**Discussion by carnun** [MichalLytek/type-graphql#1760] [Q&A]
Does anyone have an idea how you can add documentation to the schema (maybe as an SDL comment) based on the @Authorized decorators in the code? It would be fantastic for the API users if they could see what authorization they need to run a particular query ot mutation. Any suggestions or guidance would really be appreciated.

**Comment by MichalLytek** [comment]
If I were you, I would create an `@MyAuthorized()` decorators that wraps `@Authorized()` and e.g. `@Query()` decorators into single API, so that inside that unified decorator function you can generate `description` text based on the options you provide for `@Authorized()`

**Reply by carnun** [reply]
Sounds like that would work, thanks. Now I just need to figure out how the description gets used by the buildSchema function. Would my best bet be to look at the @Query() code or do you have another recommendation?

**Reply by MichalLytek** [reply]
`description` is what shows in SDL as comments, documentation

**Comment by carnun** [comment]
Here's an example of how to implement Michał's recommendation:

```
function AuthorizedMutation(roles: string[], returnTypeFunc: ReturnTypeFunc, opts?: AdvancedOptions) {
  const authorizedFn = Authorized<string>(roles)
  const rolesDescription = `AUTHORIZATION: ${roles.length ? roles.join(',') : '(all)'}`
  const options = { ...(opts || {}), description: `${opts?.description || ''}\n${rolesDescription}` }
  const mutationFn = Mutation(returnTypeFunc, options)
  return function (target: any, key: string, descriptor: any) {
    authorizedFn(target, key)
    mutationFn(target, key, descriptor)
  }
}
```
